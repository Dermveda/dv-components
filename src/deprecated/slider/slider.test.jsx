import React from 'react';
import {shallow} from 'enzyme';
import Slider from './slider.jsx';

const data = [
	{ label: 'Bad', stuff: 'stuff 1' },
	{ label: 'So-so', stuff: 'stuff 2', extra: 'hi' },
	{ label: 'Long-long-long-long-long name', stuff: 'stuff 3' },
	{ label: 'Good', stuff: 'stuff 4', other: '!!!!!' }
];

test('Creates basic range Slider', () => {
	const wrapper = shallow(
		<Slider uniqueId='test' range={6} initial={5}/>
	);
	expect(wrapper).toMatchSnapshot();
});

test('Creates basic data Slider', () => {
	const wrapper = shallow(
		<Slider uniqueId='labels'
			initial={{ label: 'So-so', stuff: 'stuff 2', extra: 'hi' }}
			data={data}
			onLabelChange={() => {}}/>
	);
	expect(wrapper).toMatchSnapshot();
});
