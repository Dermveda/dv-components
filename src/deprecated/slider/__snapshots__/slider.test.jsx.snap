// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates basic data Slider 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    id="labels_track"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
        "width": "calc(100% - 50px)",
      }
    }
  >
    <div
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "padding": "11.5px 0",
          "touchAction": "manipulation",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "background": "#599269",
            "height": "5px",
            "width": "0px",
          }
        }
      />
      <div
        style={
          Object {
            "background": "rgba(0, 0, 0, 0.2)",
            "float": "right",
            "height": "5px",
            "marginTop": "-5px",
            "width": "100%",
          }
        }
      />
      <div
        id="labels_thumb"
        style={
          Object {
            "background": "#599269",
            "borderRadius": "50%",
            "height": "15px",
            "left": "0px",
            "marginTop": "-9.5px",
            "position": "absolute",
            "touchAction": "none",
            "transform": "scale(1)",
            "transition": ".3s transform ease-in-out",
            "width": "15px",
          }
        }
      />
    </div>
    <span
      key="0"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": "none",
        }
      }
    >
      Bad
    </span>
    <span
      key="1"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": "translateX(-40%)",
        }
      }
    >
      So-so
    </span>
    <span
      key="2"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": "translateX(-40%)",
        }
      }
    >
      Long-long-long-long-long name
    </span>
    <span
      key="3"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "transform": "translateX(-60%)",
        }
      }
    >
      Good
    </span>
  </div>
</div>
`;

exports[`Creates basic range Slider 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    id="test_track"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
        "width": "calc(100% - 50px)",
      }
    }
  >
    <div
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "padding": "11.5px 0",
          "touchAction": "manipulation",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "background": "#599269",
            "height": "5px",
            "width": "0px",
          }
        }
      />
      <div
        style={
          Object {
            "background": "rgba(0, 0, 0, 0.2)",
            "float": "right",
            "height": "5px",
            "marginTop": "-5px",
            "width": "100%",
          }
        }
      />
      <div
        id="test_thumb"
        style={
          Object {
            "background": "#599269",
            "borderRadius": "50%",
            "height": "15px",
            "left": "0px",
            "marginTop": "-9.5px",
            "position": "absolute",
            "touchAction": "none",
            "transform": "scale(1)",
            "transition": ".3s transform ease-in-out",
            "width": "15px",
          }
        }
      />
    </div>
    <span
      key="0"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      1
    </span>
    <span
      key="1"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      2
    </span>
    <span
      key="2"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      3
    </span>
    <span
      key="3"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      4
    </span>
    <span
      key="4"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 4,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      5
    </span>
    <span
      key="5"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "transform": undefined,
        }
      }
    >
      6
    </span>
  </div>
</div>
`;
